avocado/plugins/diff.py

```diff
@@ -0,0 +1,387 @@
+                job2_results.extend(config_header)
+                job2_results.extend(config2)
+
+        if (('all' in args.diff_filter or 'sysinfo' in args.diff_filter) and
+           '-sysinfo' not in args.diff_filter):
+
+            sysinfo_pre1 = self._get_sysinfo(job1_dir, 'pre')
+            sysinfo_pre2 = self._get_sysinfo(job2_dir, 'pre')
+
+            if self._has_diff(sysinfo_pre1, sysinfo_pre2):
+                sysinfo_header_pre = ['\n', 'SYSINFO PRE\n']
+                job1_results.extend(sysinfo_header_pre)
+                job1_results.extend(sysinfo_pre1)
+                job2_results.extend(sysinfo_header_pre)
+                job2_results.extend(sysinfo_pre2)
+
+            sysinfo_post1 = self._get_sysinfo(job1_dir, 'post')
+            sysinfo_post2 = self._get_sysinfo(job2_dir, 'post')
+
+            if self._has_diff(sysinfo_post1, sysinfo_post2):
+                sysinfo_header_post = ['\n', 'SYSINFO POST\n']
+                job1_results.extend(sysinfo_header_post)
+                job1_results.extend(sysinfo_post1)
+                job2_results.extend(sysinfo_header_post)
+                job2_results.extend(sysinfo_post2)
+
+        job_diff = difflib.unified_diff(job1_results,
+                                        job2_results,
+                                        fromfile=job1_id,
+                                        tofile=job2_id)
+        if self.term.enabled:
+            job_diff = self._color_diff(job_diff)
+        sys.stdout.writelines(job_diff)
+
+        if (getattr(args, 'open_browser', False) and
+           getattr(args, 'html_output', None) is None):
+
+            prefix = 'avocado_diff_%s_%s_' % (job1_id[:7], job2_id[:7])
+            tmp_file = tempfile.NamedTemporaryFile(prefix=prefix,
+                                                   suffix='.html',
+                                                   delete=False)
+
+            setattr(args, 'html_output', tmp_file.name)
+
+        if getattr(args, 'html_output', None) is not None:
+            try:
+                html_diff = difflib.HtmlDiff()
+                html_diff._legend = """
+                    <table class="diff" summary="Legends">
+                    <tr> <td> <table border="" summary="Colors">
+                    <tr><th> Colors </th> </tr>
+                    <tr><td class="diff_add">&nbsp;Added&nbsp;</td></tr>
+                    <tr><td class="diff_chg">Changed</td> </tr>
+                    <tr><td class="diff_sub">Deleted</td> </tr>
+                    </table></td>
+                    <td> <table border="" summary="Links">
+                    <tr><th colspan="2"> Links </th> </tr>
+                    <tr><td>(f)irst change</td> </tr>
+                    <tr><td>(n)ext change</td> </tr>
+                    <tr><td>(t)op</td> </tr>
+                    </table></td> </tr>
+                    </table>"""
+
+                job_diff_html = html_diff.make_file(job1_results,
```

Comment:
you need to decode the string and encode it before writing, otherwise it only accepts ascii chars:

```diff
diff --git a/avocado/plugins/diff.py b/avocado/plugins/diff.py
index 8b0cadc..6634eb7 100644
--- a/avocado/plugins/diff.py
+++ b/avocado/plugins/diff.py
@@ -234,12 +234,12 @@ class Diff(CLICmd):
                     </table></td> </tr>
                     </table>"""
 
-                job_diff_html = html_diff.make_file(job1_results,
-                                                    job2_results,
+                job_diff_html = html_diff.make_file((_.decode("utf-8") for _ in job1_results),
+                                                    (_.decode("utf-8") for _ in job2_results),
                                                     fromdesc=job1_id,
                                                     todesc=job2_id)
                 with open(args.html_output, 'w') as f:
-                    f.writelines(job_diff_html)
+                    f.writelines(job_diff_html.encode("utf-8"))
 
                 sys.stdout.write('\n--\n%s\n' % args.html_output)
```